### PR TITLE
Squashed commit for ElasticSearch input, output, percolate operator.

### PR DESCRIPTION
**Pull request by yogidevendra**
Squashed commit for ElasticSearch input, output, percolate operator.

### CODE REVIEW

File: contrib/src/main/java/com/datatorrent/contrib/elasticsearch/ElasticSearchPercolatorOperator.java [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+/**
+ * Percolate operator for ElasticSearch
+ * 
+ */
+public class ElasticSearchPercolatorOperator extends BaseOperator
+{
+  public String hostName;
+  public int port;
+  public String indexName;
+  public String documentType;
```
Comment (tushargosavi) [inline contrib/src/main/java/com/datatorrent/contrib/elasticsearch/ElasticSearchPercolatorOperator.java]: missing NotNull annotations

File: contrib/src/main/java/com/datatorrent/contrib/elasticsearch/AbstractElasticSearchOutputOperator.java [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+ * 
+ * @displayName ElasticSearch Output
+ * @category Store
+ * @since 2.0.1
+ */
+public abstract class AbstractElasticSearchOutputOperator<T, S extends ElasticSearchConnectable> extends AbstractStoreOutputOperator<T, S>
+{
+  protected static final int DEFAULT_BATCH_SIZE = 1000;
+  @Min(1)
+  protected int batchSize = DEFAULT_BATCH_SIZE;
```
Comment (tushargosavi) [inline contrib/src/main/java/com/datatorrent/contrib/elasticsearch/AbstractElasticSearchOutputOperator.java]: getter and setter for batch size, for setting property from xml file

File: contrib/src/main/java/com/datatorrent/contrib/elasticsearch/AbstractElasticSearchInputOperator.java [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+   * Emit one tuple per {@code SearchHit} for given search query
+   * 
+   * @see com.datatorrent.api.InputOperator#emitTuples()
+   */
+  @Override
+  public void emitTuples()
+  {
+    SearchResponse response = getSearchRequestBuilder().execute().actionGet();
+    for (SearchHit hit : response.getHits().hits()) {
+      outputPort.emit(convertToTuple(hit));
```
Comment (tushargosavi) [inline contrib/src/main/java/com/datatorrent/contrib/elasticsearch/AbstractElasticSearchInputOperator.java]: check for null return value from convertToTuple.

**Comment by 243826** [comment]
too old!

**Pull request closed without merging**